syntax = "proto3";

package ZSS.Protocol;

import "zss_debug.proto";

message RecMessages {
	repeated RecMessage recMsgs = 1;
}

message RecMessage {
	bool ctrlC                    = 1;
	SelectedArea selectedArea     = 2;
	MaintainVision maintainVision = 3;
	repeated Debug_Msgs debugMsgs          = 4;
}

message SelectedArea {
	float maxX = 1;
	float maxY = 2;
	float minX = 3;
	float minY = 4;
}

message MaintainVision {
	int32 lastTouch         = 1;
	int32 lastTouchTeam     = 2;
	repeated TeamRobotMsg processMsg = 3;
	repeated TeamRobotMsg maintain   = 4;
	Balls4Rec balls         = 5;
}

message TeamRobotMsg {
	int32 size      = 1;
	repeated Robot4Rec robot = 2;
}

message Robot4Rec {
	int32 id     = 1;
	double posX  = 2;
	double posY  = 3;
	double angle = 4;
}

message Balls4Rec {
	int32 size    = 1;
	repeated Ball4Rec ball = 2;
}

message Ball4Rec {
	double posX = 1;
	double posY = 2;
	bool valid  = 3;
}
